### ISSUE TITLE
Unexpected Markdown Rendering Differences in Nested Components

### ISSUE THREAD

**Issue by OtokoNoIzumi** [widgetti/solara#905]
## Issue Description
I've encountered an unexpected behavior where identical Markdown content renders differently depending on its location within components. The same Markdown text appears differently when rendered directly versus when nested inside a Card component.

## Reproduction Steps
1. Create two identical Markdown components with the same content and style
2. Render one directly and one inside a Card component
3. Observe the different rendering results

## Code Example

```python

class ThemeStyles:
    LAYOUT = {
        "container": {
            "width": "100%"
        },
        "section_title": {
            "margin-bottom": "-4px",
            "padding": "0",
            "background": "pink"
        },
    }
# Direct rendering - renders as expected
solara.Markdown("## Title", style=ThemeStyles.LAYOUT["section_title"])
with solara.Card(style={'width': '100%', 'border-radius': '12px', 'padding': '0', 'cursor': 'pointer', 'position': 'relative', 'transition': 'all 0.3s ease'}):
    solara.Markdown("## some word"])
# Inside SelectionList component - renders same
@solara.component
def SelectionList(items, ...):
    solara.Markdown(f"## {title}", style=ThemeStyles.LAYOUT["section_title"])

# Inside SelectionList component - renders differently
@solara.component
def SelectionList(items, ...):
    solara.Markdown(f"## {title}", style=ThemeStyles.LAYOUT["section_title"])
    with solara.Card(style={'width': '100%', 'border-radius': '12px', 'padding': '0', 'cursor': 'pointer', 'position': 'relative', 'transition': 'all 0.3s ease'}):
        solara.Markdown("## some word"])
```
## Expected Behavior
those all become the same

## Current Behavior
![image](https://github.com/user-attachments/assets/0fc30031-f396-47a5-aa2a-3c4102500a3b)

## Specifications

  - Solara Version: 1.40.0
  - Platform: Windows
  - Affected Python Versions: 3.10.14

**Comment by iisakkirotko** [comment]
Hi @OtokoNoIzumi! I tried on PyCafe, but I couldn't replicate your issue.

Could you modify https://py.cafe/iisakkirotko/solara-issue-851 in a way so that it shows the incorrect behaviour? Afterwards you can share a link through the share menu on the site.

**Comment by OtokoNoIzumi** [comment]
> Hi @OtokoNoIzumi! I tried on PyCafe, but I couldn't replicate your issue.
> 
> Could you modify https://py.cafe/iisakkirotko/solara-issue-851 in a way so that it shows the incorrect behaviour? Afterwards you can share a link through the share menu on the site.

OH，Fixed component alignment issue by adding proper context and styling

The alignment problem was resolved by ensuring proper context within components and adding appropriate container styles:

```python
with solara.Column(
align="center",
style=ThemeStyles.LAYOUT["main_selector_container"]
):
```

Added corresponding style definition:
```python
"main_selector_container": {
"width": "100%",
"margin-bottom": "-4px",
"padding": "0", # Important: missing padding 0 can cause incorrect justify-center behavior
}
```

but there is still another problem relative?
those two section surposed to be same but there is a huge code...
![image](https://github.com/user-attachments/assets/02ecc2cf-6797-490e-be16-4f1e819e5bc1)

test code
```python

@solara.component
def SelectionList(
    items: List[Dict],
    selected_id: str,
    brand_color: str,
    on_select: Callable,
    render_price: Callable,
    title: str = "",
    subtitle: str = None,
    device_type: str = "Desktop"
):
    """通用选择列表组件"""
    with solara.Column(align="center",style=ThemeStyles.LAYOUT["main_selector_container"]):
        solara.Markdown(f"## {title}", style=ThemeStyles.LAYOUT["section_title"])
        if subtitle:
            solara.Markdown(subtitle, style=ThemeStyles.TEXT["hint_compact"])

        for item in items:
            print('test-21.2',ThemeStyles.get_card_style(item['id'] == selected_id, brand_color))
            with solara.Card(
                margin="-2px 0",
                style=ThemeStyles.get_card_style(item['id'] == selected_id, brand_color)
            ):
                if item.get('extra_text', ''):
                    print('test-22.2',item.get('extra_text', ''))
                with solara.Column(style=ThemeStyles.GENERAL["transparent_container"]):
                    with solara.Row(
                        justify="space-between",
                        style=ThemeStyles.GENERAL["transparent_container"]
                    ):
                        TitlePriceSection(
                            title=item['title'],
                            price_text=render_price(item),
                            device_type=device_type
                        )
                        DescriptionSection(
                            description=item['description'],
                            extra_text=item.get('extra_text', ''),
                            use_markdown_style=item.get('use_markdown', False)
                        )

                solara.Button(
                    "",
                    on_click=lambda i=item['id']: on_select(i),
                    style=ThemeStyles.BUTTON["transparent"]
                )


            def render_install_price(item):
                return '包含在套餐中' if item['rate'] == 0 else f'费率: {item["rate"]*100}%'

            install_items = [
                {
                    'id': install_id,
                    'title': option.get('手动设置_安装支持服务名', ''),
                    'rate': float(option.get('手动设置_产品价格费率', 0)),
                    'description': option.get('手动设置_服务内容描述', ''),
                    'extra_text': (
                        f"<br/><span style='color: #{brand_color if install_id == page_state.get_state('install_id') else '000000'}"
                        f"><b>最低服务费: {option.get('手动设置_最低服务单价', 0)}元</b></span>"
                    ) if option.get('手动设置_最低服务单价', 0) else ""
                }
                for install_id, option in data_loader.install_list.items()
            ]
            print('test-23.2',install_items)
            SelectionList(
                title="安装方案选择",
                items=install_items,
                selected_id=page_state.get_state('install_id'),
                brand_color=brand_color,
                on_select=lambda i: page_state.set_state('install_id', i),  # 修改这里
                render_price=render_install_price,
                device_type=device_type
            )

            with solara.Column(align="center"):
                solara.Markdown("## 安装方案选择", style=ThemeStyles.LAYOUT["section_title"])
                for install_id, install_option in data_loader.install_list.items():
                    install_service_name = install_option.get('手动设置_安装支持服务名', '')
                    install_rate = float(install_option.get('手动设置_产品价格费率', 0))
                    install_min = install_option.get('手动设置_最低服务单价', 0)
                    install_info = install_option.get('手动设置_服务内容描述', "")

                    card_style = ThemeStyles.get_card_style(install_id == page_state.get_state('install_id'), brand_color)
                    price_text = '包含在套餐中' if install_rate == 0 else f'费率: {install_rate*100}%'
                    min_fee_text = "" if install_min == 0 else f"<br/><span style='color: #{brand_color if install_id == page_state.get_state('install_id') else '000000'}'><b>最低服务费: {install_min}元</b></span>"
                    with solara.Card(margin="-2px 0", style=card_style):
                        with solara.Column(style=ThemeStyles.GENERAL["transparent_container"]):
                            with solara.Row(justify="space-between", style=ThemeStyles.GENERAL["transparent_container"]):
                                TitlePriceSection(install_service_name, price_text, device_type)
                                DescriptionSection(install_info, min_fee_text)

                        solara.Button(
                            "",
                            on_click=lambda i=install_id: page_state.set_state('install_id', i),
                            style=ThemeStyles.BUTTON["transparent"]
                        )

```

print result for debug still the same but looks different

test-21.2 {'width': '100%', 'border-radius': '12px', 'padding': '0', 'cursor': 'pointer', 'position': 'relative', 'transition': 'all 0.3s ease'}
test-21.2 {'width': '100%', 'border-radius': '12px', 'padding': '0', 'cursor': 'pointer', 'position': 'relative', 'transition': 'all 0.3s ease', 'background': 'linear-gradient(145deg, rgba(255,107,0,0.08) 0%, rgba(255,107,0,0.16) 100%)', 'border': '1px solid rgba(255,107,0,0.2)', 'box-shadow': '0 0 20px rgba(255,107,0,0.15)'}
test-22.2 <br/><span style='color: #FF6B00><b>最低服务费: 500元</b></span>

test-23.2 [{'id': 'Install001', 'title': 'DIY自行安装', 'rate': 0.0, 'description': '随产品免费提供产品说明书与安装录像\n享受DIY的安装乐趣并掌控您的设备\n售后保障时无法享受套餐优惠', 'extra_text': ''}, {'id': 'Install002', 'title': '专业上门安装', 'rate': 0.25, 'description': '专业人员上门安装调试\n一年内赠予一次免费上门售后排障\n更多售后服务时可享折扣优惠', 'extra_text': "<br/><span style='color: #FF6B00><b>最低服务费: 500元</b></span>"}, {'id': 'Install003', 'title': '全程监工安装', 'rate': 0.35, 'description': '从施工期开始多次监工安装（至少三次）\n一年内赠予一次免费上门售后排障\n更多售后服务时可享折扣优惠', 'extra_text': "<br/><span style='color: #000000><b>最低服务费: 1000元</b></span>"}, {'id': 'Install004', 'title': '智能场景专家安装', 'rate': 0.4, 'description': '最专业的设备安装\n根据顾客生活特点定制化设计智能场景\n一年内赠予两次免费上门售后排障\n更多售后服务时可享折扣优惠', 'extra_text': "<br/><span style='color: #000000><b>最低服务费: 3000元</b></span>"}]

**Comment by OtokoNoIzumi** [comment]
> Hi @OtokoNoIzumi! I tried on PyCafe, but I couldn't replicate your issue.
> 
> Could you modify https://py.cafe/iisakkirotko/solara-issue-851 in a way so that it shows the incorrect behaviour? Afterwards you can share a link through the share menu on the site.
Done:
https://py.cafe/iisakkirotko/solara-issue-851
![image](https://github.com/user-attachments/assets/5cb0ccb6-bc2d-43ac-b9f0-ffe26332cf81)

**Comment by iisakkirotko** [comment]
Hi @OtokoNoIzumi! Thanks for the minimal example! I looked at it, and this is a side-effect of Solara's expected behaviour. Essentially currently, if there are two Solara components without a parent, like `solara.Markdown` containing the title and `solara.Card` in this component:

```python
@solara.component
def SelectionList():
    solara.Markdown(f"## Title1")
    with solara.Card(
        margin="-2px 0",style={...}):
        solara.Markdown("## some word")
```

What we do currently is wrap them in a `solara.Column`, because Reacton used to not support directly adding multiple children with no container. We plan to fix this in Solara 2.0, since removing the wrapping `Column` will be a breaking change.

See this issue to track our progress: https://github.com/widgetti/solara/issues/596

In the meantime, you can make the cards take up the full width of the page by manually wrapping them in a `solara.Column`, with `style={"width": "100%", "align-items": "stretch"}`, a `solara.Div` with `style_="width: 100%;"`, or a different Solara component like in https://py.cafe/iisakkirotko/solara-issue-851-1

**Comment by OtokoNoIzumi** [comment]
> > Hi @OtokoNoIzumi! I tried on PyCafe, but I couldn't replicate your issue.
> > Could you modify https://py.cafe/iisakkirotko/solara-issue-851 in a way so that it shows the incorrect behaviour? Afterwards you can share a link through the share menu on the site.
> 
> OH，Fixed component alignment issue by adding proper context and styling
> 
> The alignment problem was resolved by ensuring proper context within components and adding appropriate container styles:
> 
> ```python
> with solara.Column(
> align="center",
> style=ThemeStyles.LAYOUT["main_selector_container"]
> ):
> ```
> 
> Added corresponding style definition:
> 
> ```python
> "main_selector_container": {
> "width": "100%",
> "margin-bottom": "-4px",
> "padding": "0", # Important: missing padding 0 can cause incorrect justify-center behavior
> }
> ```
> 
> but there is still another problem relative? those two section surposed to be same but there is a huge code... ![image](https://private-user-images.githubusercontent.com/29002083/392620424-02ecc2cf-6797-490e-be16-4f1e819e5bc1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM0ODMwMDAsIm5iZiI6MTczMzQ4MjcwMCwicGF0aCI6Ii8yOTAwMjA4My8zOTI2MjA0MjQtMDJlY2MyY2YtNjc5Ny00OTBlLWJlMTYtNGYxZTgxOWU1YmMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA2VDEwNTgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4YTAzNmIzY2JjY2FmZjliYWIxM2ExYjYyNzMzNjkyZDdkNDBiNTk0OTU4NDc5NjkzZDgzZmIxMDNiZTljMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mZ5Ivwl31r7RqyUp0S4eCfIe8Ceddh6doAw3y6UgYQs)
> 
> test code
> 
> ```python
> @solara.component
> def SelectionList(
>     items: List[Dict],
>     selected_id: str,
>     brand_color: str,
>     on_select: Callable,
>     render_price: Callable,
>     title: str = "",
>     subtitle: str = None,
>     device_type: str = "Desktop"
> ):
>     """通用选择列表组件"""
>     with solara.Column(align="center",style=ThemeStyles.LAYOUT["main_selector_container"]):
>         solara.Markdown(f"## {title}", style=ThemeStyles.LAYOUT["section_title"])
>         if subtitle:
>             solara.Markdown(subtitle, style=ThemeStyles.TEXT["hint_compact"])
> 
>         for item in items:
>             print('test-21.2',ThemeStyles.get_card_style(item['id'] == selected_id, brand_color))
>             with solara.Card(
>                 margin="-2px 0",
>                 style=ThemeStyles.get_card_style(item['id'] == selected_id, brand_color)
>             ):
>                 if item.get('extra_text', ''):
>                     print('test-22.2',item.get('extra_text', ''))
>                 with solara.Column(style=ThemeStyles.GENERAL["transparent_container"]):
>                     with solara.Row(
>                         justify="space-between",
>                         style=ThemeStyles.GENERAL["transparent_container"]
>                     ):
>                         TitlePriceSection(
>                             title=item['title'],
>                             price_text=render_price(item),
>                             device_type=device_type
>                         )
>                         DescriptionSection(
>                             description=item['description'],
>                             extra_text=item.get('extra_text', ''),
>                             use_markdown_style=item.get('use_markdown', False)
>                         )
> 
>                 solara.Button(
>                     "",
>                     on_click=lambda i=item['id']: on_select(i),
>                     style=ThemeStyles.BUTTON["transparent"]
>                 )
> 
> 
>             def render_install_price(item):
>                 return '包含在套餐中' if item['rate'] == 0 else f'费率: {item["rate"]*100}%'
> 
>             install_items = [
>                 {
>                     'id': install_id,
>                     'title': option.get('手动设置_安装支持服务名', ''),
>                     'rate': float(option.get('手动设置_产品价格费率', 0)),
>                     'description': option.get('手动设置_服务内容描述', ''),
>                     'extra_text': (
>                         f"<br/><span style='color: #{brand_color if install_id == page_state.get_state('install_id') else '000000'}"
>                         f"><b>最低服务费: {option.get('手动设置_最低服务单价', 0)}元</b></span>"
>                     ) if option.get('手动设置_最低服务单价', 0) else ""
>                 }
>                 for install_id, option in data_loader.install_list.items()
>             ]
>             print('test-23.2',install_items)
>             SelectionList(
>                 title="安装方案选择",
>                 items=install_items,
>                 selected_id=page_state.get_state('install_id'),
>                 brand_color=brand_color,
>                 on_select=lambda i: page_state.set_state('install_id', i),  # 修改这里
>                 render_price=render_install_price,
>                 device_type=device_type
>             )
> 
>             with solara.Column(align="center"):
>                 solara.Markdown("## 安装方案选择", style=ThemeStyles.LAYOUT["section_title"])
>                 for install_id, install_option in data_loader.install_list.items():
>                     install_service_name = install_option.get('手动设置_安装支持服务名', '')
>                     install_rate = float(install_option.get('手动设置_产品价格费率', 0))
>                     install_min = install_option.get('手动设置_最低服务单价', 0)
>                     install_info = install_option.get('手动设置_服务内容描述', "")
> 
>                     card_style = ThemeStyles.get_card_style(install_id == page_state.get_state('install_id'), brand_color)
>                     price_text = '包含在套餐中' if install_rate == 0 else f'费率: {install_rate*100}%'
>                     min_fee_text = "" if install_min == 0 else f"<br/><span style='color: #{brand_color if install_id == page_state.get_state('install_id') else '000000'}'><b>最低服务费: {install_min}元</b></span>"
>                     with solara.Card(margin="-2px 0", style=card_style):
>                         with solara.Column(style=ThemeStyles.GENERAL["transparent_container"]):
>                             with solara.Row(justify="space-between", style=ThemeStyles.GENERAL["transparent_container"]):
>                                 TitlePriceSection(install_service_name, price_text, device_type)
>                                 DescriptionSection(install_info, min_fee_text)
> 
>                         solara.Button(
>                             "",
>                             on_click=lambda i=install_id: page_state.set_state('install_id', i),
>                             style=ThemeStyles.BUTTON["transparent"]
>                         )
> ```
> 
> print result for debug still the same but looks different
> 
> test-21.2 {'width': '100%', 'border-radius': '12px', 'padding': '0', 'cursor': 'pointer', 'position': 'relative', 'transition': 'all 0.3s ease'} test-21.2 {'width': '100%', 'border-radius': '12px', 'padding': '0', 'cursor': 'pointer', 'position': 'relative', 'transition': 'all 0.3s ease', 'background': 'linear-gradient(145deg, rgba(255,107,0,0.08) 0%, rgba(255,107,0,0.16) 100%)', 'border': '1px solid rgba(255,107,0,0.2)', 'box-shadow': '0 0 20px rgba(255,107,0,0.15)'} test-22.2 <span style='color: #FF6B00>最低服务费: 500元
> 
> test-23.2 [{'id': 'Install001', 'title': 'DIY自行安装', 'rate': 0.0, 'description': '随产品免费提供产品说明书与安装录像\n享受DIY的安装乐趣并掌控您的设备\n售后保障时无法享受套餐优惠', 'extra_text': ''}, {'id': 'Install002', 'title': '专业上门安装', 'rate': 0.25, 'description': '专业人员上门安装调试\n一年内赠予一次免费上门售后排障\n更多售后服务时可享折扣优惠', 'extra_text': "<span style='color: #FF6B00>最低服务费: 500元"}, {'id': 'Install003', 'title': '全程监工安装', 'rate': 0.35, 'description': '从施工期开始多次监工安装（至少三次）\n一年内赠予一次免费上门售后排障\n更多售后服务时可享折扣优惠', 'extra_text': "<span style='color: #000000>最低服务费: 1000元"}, {'id': 'Install004', 'title': '智能场景专家安装', 'rate': 0.4, 'description': '最专业的设备安装\n根据顾客生活特点定制化设计智能场景\n一年内赠予两次免费上门售后排障\n更多售后服务时可享折扣优惠', 'extra_text': "<span style='color: #000000>最低服务费: 3000元"}]

thx for replying ,waiting for a breaking new version! 
BTW what happened here? is these something same like wraped column? i saw window.solara.cdn ...
